### PR TITLE
Melissa/842 error audit fix low hanging fruit and loose ends

### PR DESCRIPTION
**Pull request by mmnoo**
See code comments and descriptive commit messages

Leave testing for Kim end crew

### CODE REVIEW

File: src/App/integrationTests/managementRegime/App.managementRegimeCreateOffline.test.js [hunk 1]
```diff
@@ -159,9 +162,7 @@ describe('Offline', () => {
     expect(await screen.findByTestId('management-regime-toast-error')).toHaveTextContent(
       'The management regime failed to save both on your computer and in the MERMAID online system.',
     )
-    expect(await screen.findByTestId('management-regime-toast-error')).toHaveTextContent(
-      'this is a dexie error',
-    )
```
Comment (mmnoo) [inline src/App/integrationTests/managementRegime/App.managementRegimeCreateOffline.test.js:164]: removing an overwhelm of multiple toasts (yes this kind of is another ticket, but its related). Showing the above assertion (ln 162) is sufficient for a user who doesnt care about dexie or code details.

File: src/App/integrationTests/managementRegime/App.managementRegimeCreateOffline.test.js [hunk 2]
```diff
@@ -159,9 +162,7 @@ describe('Offline', () => {
...
+    expect(consoleSpy).toHaveBeenCalledWith(dexieError)
```
Comment (mmnoo) [inline src/App/integrationTests/managementRegime/App.managementRegimeCreateOffline.test.js:165]: we just want to console that instead.

File: src/components/pages/ManagementRegime/ManagementRegime.js [hunk 3]
```diff
@@ -352,11 +352,14 @@ const ManagementRegime = ({ isNewManagementRegime }) => {
           }
 
           if (!isAppOnline) {
-            showSyncToastError({
```
Comment (mmnoo) [inline src/components/pages/ManagementRegime/ManagementRegime.js:355]: oops, if we are offline, there will be no sync related toasts. Just show a normal toast.  And console that error.

File: src/components/pages/ManagementRegime/ManagementRegime.js [hunk 4]
```diff
@@ -406,6 +409,7 @@ const ManagementRegime = ({ isNewManagementRegime }) => {
   )
 
   const deleteRecord = () => {
+    // only available online
```
Comment (mmnoo) [inline src/components/pages/ManagementRegime/ManagementRegime.js:412]: simplifying some of the logic that existed for a feature that doesnt exist (offline deleting)

File: src/components/pages/Site/Site.js [hunk 5]
```diff
@@ -352,11 +352,14 @@ const Site = ({ isNewSite }) => {
             })
           }
           if (!isAppOnline) {
-            showSyncToastError({
```
Comment (mmnoo) [inline src/components/pages/Site/Site.js:355]: again, if we're offline, we wont be syncing, so just show a normal toast, not the special sync toast

File: src/components/pages/collectRecordFormPages/CollectRecordFormPage/CollectRecordFormPage.js [hunk 6]
```diff
@@ -313,7 +313,7 @@ const CollectRecordFormPage = ({
           handleHttpResponseError({
             error,
             callback: () => {
-              toast.error(...getToastArguments(language.error.collectRecordSubmit))
+              toast.error(...getToastArguments(language.error.collectRecordValidationIgnore))
```
Comment (mmnoo) [inline src/components/pages/collectRecordFormPages/CollectRecordFormPage/CollectRecordFormPage.js:316]: fixed wrong error message being used

File: src/language.js [hunk 7]
```diff
@@ -19,8 +19,8 @@ const error = {
   503: 'Server error: please contact support@datamermaid.org',
   apiDataSync: 'The app was not able to sync data with the API. Please try again.',
   appNotAuthenticatedOrReady: 'Initialization error. Try reloading or reauthenticating.',
-  collectRecordChoicesUnavailable:
-    'Options data for creating a sample unit is currently unavailable. Please try again',
+  collectRecordSupportingDataUnavailable:
+    'Supporting data for creating a sample unit is currently unavailable. Please try again',
```
Comment (mmnoo) [inline src/language.js:23]: I dont know how this weird language/logic evolved, but we/I copied and pasted its use for all the collect recorrds. oof

File: src/library/handleHttpResponseError.js [hunk 8]
```diff
@@ -3,19 +3,13 @@ import language from '../language'
...
 
 const handleHttpResponseError = ({ error, callback, logoutMermaid }) => {
-  const errorStatus = error?.response?.status
-
-  if (!errorStatus) {
-    // If there is a general error which does not have a response status
-    // Don't throw an error. This should only ever encountered if axios is called after the the test is destroyed
-    toast.error(...getToastArguments(language.error.generic))
+  if (error) {
     console.error(error)
```
Comment (mmnoo) [inline src/library/handleHttpResponseError.js:7]: we're just going to simplify here and console out any error if it is defined

File: src/library/handleHttpResponseError.js [hunk 9]
```diff
@@ -28,6 +22,8 @@ const handleHttpResponseError = ({ error, callback, logoutMermaid }) => {
         'A 401 error occurred. handleHttpResponseError requires a logoutMermaid function in its config object.',
       )
     }
+
+    return
```
Comment (mmnoo) [inline src/library/handleHttpResponseError.js:26]: this used to return in the next if statement implicitly. I removed 401 from being part of the next if statement and returned explicitly in the 401-specific if statement.

File: src/library/handleHttpResponseError.js [hunk 10]
```diff
@@ -45,9 +41,6 @@ const handleHttpResponseError = ({ error, callback, logoutMermaid }) => {
 
     return
   }
-
-  // Display a generic error if we haven't returned yet
-  toast.error(...getToastArguments(language.error.generic))
```
Comment (mmnoo) [inline src/library/handleHttpResponseError.js:50]: We were being too liberal on the toasting of generic error messages. This http error handler utility is just used to intercept some http errors, if none are intercepted, it doesnt mean we should fallback within the utility to toast out some generic blabber (we always have the option to do that with the passed in callback)

**Comment by mmnoo** [comment]
@saanobhaai just FYI, this is some loose ends and low hanging fruit from my audit of vague error messages. YOu can also see the things I worked on in that document which I will link to from the trello ticket.

**Comment by mmnoo** [comment]
[audit cryptic toast error messages](https://trello.com/c/k4O2MjrT/842-audit-cryptic-toast-error-messages)

File: src/language.js (hunk 7 above)
Comment (julia-rom) [reply src/language.js:23]: makes more sense!

File: src/library/handleHttpResponseError.js (hunk 8 above)
Comment (julia-rom) [reply src/library/handleHttpResponseError.js:7]: are we getting rid of a toast in this case? how will the user know if there is an http error?

File: src/library/handleHttpResponseError.js (hunk 10 above)
Comment (julia-rom) [reply src/library/handleHttpResponseError.js:50]: nice catch! agreed

**Review by julia-rom — APPROVED** [review]
code looks good, just wanted to clarify removing a toast for handleHttpResponseError (in comments)

File: src/library/handleHttpResponseError.js (hunk 8 above)
Comment (mmnoo) [reply src/library/handleHttpResponseError.js:7]: thats funther on down the file. Toasts will happen for a subset of http error types still and if there is a callback, that will be called to handle any other errors after

File: src/library/handleHttpResponseError.js [hunk 11]
```diff
@@ -28,10 +20,14 @@ const handleHttpResponseError = ({ error, callback, logoutMermaid }) => {
         'A 401 error occurred. handleHttpResponseError requires a logoutMermaid function in its config object.',
       )
     }
+
+    return
   }
 
+  const otherErrorStatusesToRespondTo = [403, 500, 502, 503]
+
```
[the remark this answers is not in the document]
Comment (julia-rom) [reply src/library/handleHttpResponseError.js:28]: ok got it- thanks for clarifying! all good my end code-wise

**Pull request merged**